src/device.rs

```diff
@@ -51,6 +54,38 @@ impl Device {
         }))
     }
 
+    /// Encrypt an event to be sent to this device.
+    /// (olm encryption).
+    ///
+    /// Prior to calling this method you must ensure that an olm session is
+    /// available for the target device. This can be done by calling the
+    /// [`get_missing_sessions()`](OlmMachine::get_missing_sessions)
+    ///
+    /// The caller is responsible for sending the encrypted
+    /// event to the target device. If multiple messages are
+    /// encrypted using this method they should be sent in the same order as
+    /// they are encrypted.
+    ///
+    /// # Returns
+    /// Returns a Promise for a `json` string of the encrypted event.
```

Comment:
Finally found the way to do it (needed to use `Serializer::json_compatible()`)

Reply:
OH! Sorry. When I asked this, I hadn't realised that `encrypt_event_raw` returns JSON. I don't think it's correct that we then deserialize that object back into a rust-side `ToDeviceEncryptedEventContent`, and then re-serialize *that* back to a js object.

Since we have JSON, let's just return it and deal with it on the JS side. Sorry!

Reply:
[Arguably it seems a shame that `encrypt_event_raw` serializes its response, but that's a sidebar]